tests/unit-tests/internal/student-progress/lesson-progress/repositories/test-class-tables-based-lesson-progress-repository.php

```diff
@@ -0,0 +1,305 @@
+<?php
+
+namespace SenseiTest\Internal\Student_Progress\Lesson_Progress\Repositories;
+
+use DateTimeImmutable;
+use Sensei\Internal\Student_Progress\Lesson_Progress\Models\Lesson_Progress;
+use Sensei\Internal\Student_Progress\Lesson_Progress\Repositories\Tables_Based_Lesson_Progress_Repository;
+use Sensei_Course;
+use wpdb;
+
+/**
+ * Tests for Tables_Based_Lesson_Progress_Repository.
+ *
+ * @covers \Sensei\Internal\Student_Progress\Lesson_Progress\Repositories\Tables_Based_Lesson_Progress_Repository
+ */
+class Tables_Based_Lesson_Progress_Repository_Test extends \WP_UnitTestCase {
+	public function testCreate_ParamsGiven_InsertsToWpdb(): void {
+		/* Arrange. */
+		$wpdb       = $this->createMock( wpdb::class );
+		$repository = new Tables_Based_Lesson_Progress_Repository( $wpdb );
+
+		/* Expect & Act. */
+		$wpdb
+			->expects( $this->once() )
+			->method( 'insert' )
+			->with(
+				'sensei_lms_progress',
+				$this->callback(
+					function( $array ) {
+						if ( ! isset( $array['post_id'] ) || 1 !== $array['post_id'] ) {
+							return false;
+						}
+						if ( ! isset( $array['user_id'] ) || 2 !== $array['user_id'] ) {
+							return false;
+						}
+						if ( ! array_key_exists( 'parent_post_id', $array ) || ! is_null( $array['parent_post_id'] ) ) {
+							return false;
+						}
+						if ( ! isset( $array['type'] ) || 'lesson' !== $array['type'] ) {
+							return false;
+						}
+						if ( ! isset( $array['status'] ) || 'in-progress' !== $array['status'] ) {
+							return false;
+						}
+						return true;
+					}
+				),
+				[
+					'%d',
+					'%d',
+					null,
+					'%s',
+					'%s',
+					'%s',
+					null,
+					'%s',
+					'%s',
+				]
+			);
+		$repository->create( 1, 2 );
+	}
+
+	public function testCreate_ParamsGiven_ReturnsMatchingCourseProgress(): void {
+		/* Arrange. */
+		$wpdb            = $this->createMock( wpdb::class );
+		$wpdb->insert_id = 3;
+		$repository      = new Tables_Based_Lesson_Progress_Repository( $wpdb );
+
+		/* Act. */
+		$progress = $repository->create( 1, 2 );
+
+		/* Assert. */
+		$expected = [
+			'id'        => 3,
+			'lesson_id' => 1,
+			'user_id'   => 2,
+			'status'    => 'in-progress',
+		];
+		self::assertSame( $expected, $this->export_progress( $progress ) );
+	}
+
+	public function testGet_NotFound_ReturnsNull(): void {
+		/* Arrange. */
+		$wpdb = $this->createMock( wpdb::class );
+		$wpdb
+			->method( 'get_row' )
+			->willReturn( null );
+		$repository = new Tables_Based_Lesson_Progress_Repository( $wpdb );
+
+		/* Act. */
+		$progress = $repository->get( 1, 2 );
+
+		/* Assert. */
+		self::assertNull( $progress );
+	}
+
+	public function testGet_ProgressFound_ReturnsMatchingProgress(): void {
+		/* Arrange. */
+		$wpdb = $this->createMock( wpdb::class );
```

Comment:
Do you think it's also a good idea to have integration tests for methods with custom SQL? Currently, if I update the [select SQL statement](https://github.com/Automattic/sensei/pull/5894/files#diff-7af5a4e7f9b304e889985b0be06cfadbf99b8467a1deb6a2e23572bdeaaee48aR109) so it won't return results, the tests are still passing.

Reply:
You're right, we need a separate integration test for that 👍

Reply:
Added integration tests for those methods here: https://github.com/Automattic/sensei/pull/5894/commits/cf2fbbf678d7029b8d337f601a9052318da641e6